go-controller/pkg/clustermanager/routeadvertisements/controller.go

```diff
@@ -994,6 +984,44 @@ func (c *Controller) updateRAStatus(ra *ratypes.RouteAdvertisements, hadUpdates
 	return nil
 }
 
+func (c *Controller) getSelectedNADs(networkSelectors apitypes.NetworkSelectors) ([]*nadtypes.NetworkAttachmentDefinition, error) {
+	var selected []*nadtypes.NetworkAttachmentDefinition
+	for _, networkSelector := range networkSelectors {
+		switch networkSelector.NetworkSelectionType {
+		case apitypes.DefaultNetwork:
+			// if we are selecting the default networkdefault network label,
+			// make sure a NAD exists for it
+			nad, err := c.getOrCreateDefaultNetworkNAD()
+			if err != nil {
+				return nil, fmt.Errorf("failed to get/create default network NAD: %w", err)
+			}
+			selected = append(selected, nad)
+		case apitypes.ClusterUserDefinedNetworks:
+			nadSelector, err := metav1.LabelSelectorAsSelector(&networkSelector.ClusterUserDefinedNetworkSelector.NetworkSelector)
+			if err != nil {
+				return nil, err
+			}
+			nads, err := c.nadLister.List(nadSelector)
+			if err != nil {
+				return nil, err
+			}
+			for _, nad := range nads {
+				// check this NAD is controlled by a CUDN
+				controller := metav1.GetControllerOfNoCopy(nad)
+				isCUDN := controller != nil && controller.Kind == cudnController.Kind && controller.APIVersion == cudnController.GroupVersion().String()
+				if !isCUDN {
+					continue
+				}
+				selected = append(selected, nad)
```

Comment:
Shouldn't we also exclude any NADs that do not represent a primary network?

Reply:
good point, as well as localnet types?

Reply:
That filtering is already done in `generateFRRConfigurations`. Filtering applied in this method only corresponds to the what the selector itself is telling us. This potentially can be common code in the future for any APIs using the selector.

Reply:
Aside from that, if they select something that we don't support, we return error `generateFRRConfigurations`, we don't ignore it.